### DISCUSSION TITLE
AdGuard - CloudFlare - DoH sporadically Yes via 1.1.1.1/help

### DISCUSSION THREAD

**Discussion by eltonajmenezes** [trinib/Adguard-Wireguard-Unbound-Cloudflare#48] [Help]
### Operating System

64-bit

### Project

Cloudflare

### Platform

Mac, IOS

### Browser

Chrome, Other

### Issue

Not working, Other (explain in description)

### Issue Description

The first time you setup Cloudflare and point AdGuard to it; there seems to be no issue at all.
Even the 1.1.1.1/help test always is consistent and indicated Yes for DoH and DoT.

But if a restart or shutdown happens then for some reason the status of clodflared when checked in terminal shows the following 

`flared[609]: 2022-04-30T10:48:21Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:21 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:21Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:21 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:21Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:21 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:21Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:21 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:21Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:21 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:21Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:21 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:21Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:22 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:22Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:18:22 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:48:22Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile>
Apr 30 16:29:14 eltonsraspberrypi cloudflared[609]: 2022-04-30T10:59:14Z ERR failed to connect to an HTTPS backend "https://1.1.1.1/dns-query" error="faile> `

These warnings can be 3 - 10 in number. 
If you stop the clodfalred service and restart it; the Warning goes away. But eventually it reappears.

Now the odd part is that around the same time this happens; the 1.1.1.1/help ,test gives you sporadically Yes for DoH when tested in Safari and Chrome browser compared to initially and either way the cache was cleared.

![A9420696-468E-41C4-9374-99BB061C8902](https://user-images.githubusercontent.com/66936537/166317194-564cd3e5-48f0-481c-b707-77853764b6d9.jpeg)

https://user-images.githubusercontent.com/66936537/166317177-85b8b5f0-7461-4142-9c6c-0ea051c4a504.MOV

![FA467290-7CD4-4121-81B3-B36D61B9CADD](https://user-images.githubusercontent.com/66936537/166317160-1f81048b-317b-493d-86b3-e47726926011.jpeg)

I have searched multiple forums but no one’s solution seems to stick.

Also moving away from “Parallel Requests” to “Fastest IP Address” the  issue goes but the speed of fetching web content is reduced by a small amount.

**Comment by trinib** [comment]
I'll have to test and see.. is it raspberry pi 64bit? you are using? cause 64bit came out February and could have issues.. if you do try 32bit and see if you get same errors

or it could be cloudflare version

**Comment by eltonajmenezes** [comment]
> I'll have to test and see.. is it raspberry pi 64bit? you are using? cause 64bit came out February and could have issues.. if you do try 32bit and see if you get same errors
> 
> or it could be cloudflare version 

Yes I was using 32Bit and this issue was present, moved to 64 Bit and the issue still happens 

I am using the latest clodflared version present FYI

**Comment by eltonajmenezes** [comment]
I realized one more thing.
Immediately after a restart the service seems to be fine and the behavior of 1.1.1.1/help DoH is also ok

**Comment by eltonajmenezes** [comment]
> @eltonajmenezes I get no errors after reboot on Raspberry OS 64bit. and I  saw no issues like yours on https://github.com/cloudflare/cloudflared
> 
> 
> 
> ![zero](https://user-images.githubusercontent.com/18756975/166457122-bb4193df-a07f-47f6-b493-f13fce9dddd7.png)
> 
> 
> 
> 

It's doesn't happen immediately. It happens all of a sudden, I will keep an eye and report back to you

**Comment by trinib** [comment]
@eltonajmenezes  sorry I now saw the issues on https://github.com/cloudflare/cloudflared/issues/91 and https://github.com/cloudflare/cloudflared/issues/306.. I guess its something i cannot fix. I tried rebooting shutting down pi and rebooting router but issue do not shows for me still..

**Comment by trinib** [comment]
@eltonajmenezes  use this method i suggested in https://github.com/trinib/AdGuard-WireGuard-Unbound-Cloudflare/issues/28#issuecomment-1115723760 for DoH. I guess I can add this to wiki as an alternative for Cloudflared tunnel client. 

[DNSCrypt](https://dnscrypt.info/) - a DNS(DoH) proxy client.

**Comment by eltonajmenezes** [comment]
> @eltonajmenezes  use this method i suggested in https://github.com/trinib/AdGuard-WireGuard-Unbound-Cloudflare/issues/28#issuecomment-1115723760 for DoH. I guess I can add this to wiki as an alternative for Cloudflared tunnel client. 
> 
> 
> 
> [DNSCrypt](https://dnscrypt.info/) - a DNS(DoH) proxy client.



I will try and replicate this and see what the dependencies are on my instance.

Maybe something is interfering. I will get back to you on this in a few days.

**Comment by trinib** [comment]
@eltonajmenezes I saw this person said they found a fix https://github.com/cloudflare/cloudflared/issues/306#issuecomment-989524710, try it and see if it works for you 

open:
```
sudo nano /etc/default/cloudflared
```

add :  `--max-upstream-conns 50`
![zero](https://user-images.githubusercontent.com/18756975/166520285-810e4a15-3ab7-4f1a-b13b-e5deaceb9713.png)

**Comment by eltonajmenezes** [comment]
> @eltonajmenezes I saw this person said they found a fix https://github.com/cloudflare/cloudflared/issues/306#issuecomment-989524710, try it and see if it works for you 
> 
> 
> 
> open:
> 
> ```
> 
> sudo nano /etc/default/cloudflared
> 
> ```
> 
> 
> 
> add :  `--max-upstream-conns 50`
> 
> ![zero](https://user-images.githubusercontent.com/18756975/166520285-810e4a15-3ab7-4f1a-b13b-e5deaceb9713.png)
> 
> 
> 
> 


@trinib 
Yes I did try this in fact but it did not help at all. 


I think I have narrowed down a possibility but need to observe and check.

**Comment by trinib** [comment]
it seems this issue happens for some and not everyone, some say it's a isp and router issue. https://github.com/cloudflare/cloudflared/issues/91#issuecomment-499912858

See if this works https://github.com/cloudflare/cloudflared/issues/91#issuecomment-491062486

**Comment by eltonajmenezes** [comment]
> it seems this issue happens for some and not everyone, some say it's a isp and router issue. https://github.com/cloudflare/cloudflared/issues/91#issuecomment-499912858
> 
> 
> 
> See if this works https://github.com/cloudflare/cloudflared/issues/91#issuecomment-491062486


Doubt it's an ISP issue, and I do not have the hosts directory on my system.

**Comment by trinib** [comment]
> > it seems this issue happens for some and not everyone, some say it's a isp and router issue. [cloudflare/cloudflared#91 (comment)](https://github.com/cloudflare/cloudflared/issues/91#issuecomment-499912858)
> > See if this works [cloudflare/cloudflared#91 (comment)](https://github.com/cloudflare/cloudflared/issues/91#issuecomment-491062486)
> 
> Doubt it's an ISP issue, and I do not have the hosts directory on my system.

you can create it .

**Comment by trinib** [comment]
This issue with cloudflare is really weird .. Hear this I have a good idea .. to really see if it's cloudfared or just your location isp router etc .. I want you to try on a VPS !!!. here is my personal referral link for free $35-https://www.vultr.com/?ref=9113990-8H
 for first user signup(limited). or if it do not work here is regular referral for $10-https://www.vultr.com/?ref=9113188

let me know if you need help setting it up .. it pretty easy and quick .. watch a youtube guide if issue

**Comment by trinib** [comment]
https://user-images.githubusercontent.com/18756975/166669805-4e5748ea-78fd-4906-8993-7df42607e920.mp4

**Comment by eltonajmenezes** [comment]
Thank you, will look into this and revert shortly. So  you want me to test this on a virtual server right?

**Comment by trinib** [comment]
> Thank you, will look into this and revert shortly. So  you want me to test this on a virtual server right?

yes

**Comment by eltonajmenezes** [comment]
After using DNScrypt running on port 5335 as well I have started to see this issue where the upstream server sporadically works and stops most of the time.

Even after restarting the service I noticed this error wouldn't go away.
`May 12 18:11:37 eltonsraspberrypi dnscrypt-proxy[546859]: [2022-05-12 18:11:37] [ERROR] Get "https://dns.cloudflare.com/dns-query?dns=yv4BAAABAAAAAAABAAACAAEAACkQAAAAAAAAFAAMABDnWpigWoLlJKOn36NBJY-N": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

So I stopped the service, changed the port again to 53000 and started it.

but again it stopped
I even did a test in this manner and I got a reply
curl portquiz.net:5335 --connect-timeout 1

But when I tested this way
dig sigfail.verteiltesysteme.net @127.0.0.1 -p 5335

 I got a response  ; <<>> DiG 9.16.27-Debian <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 5335
;; global options: +cmd
;; connection timed out; no servers could be reached

**Comment by trinib** [comment]
> After using DNScrypt running on port 5335 as well I have started to see this issue where the upstream server sporadically works and stops most of the time.
> 
> Even after restarting the service I noticed this error wouldn't go away. `May 12 18:11:37 eltonsraspberrypi dnscrypt-proxy[546859]: [2022-05-12 18:11:37] [ERROR] Get "https://dns.cloudflare.com/dns-query?dns=yv4BAAABAAAAAAABAAACAAEAACkQAAAAAAAAFAAMABDnWpigWoLlJKOn36NBJY-N": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`
> 
> So I stopped the service, changed the port again to 53000 and started it.
> 
> but again it stopped I even did a test in this manner and I got a reply curl portquiz.net:5335 --connect-timeout 1
> 
> But when I tested this way dig sigfail.verteiltesysteme.net @127.0.0.1 -p 5335
> 
> I got a response ; <<>> DiG 9.16.27-Debian <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 5335 ;; global options: +cmd ;; connection timed out; no servers could be reached

You need to show your configurations ..

**Comment by trinib** [comment]
> for me i like to share my result unbound , cloudflare-proxy , dnscrypt-proxy working fine (form me i use family upstream you can use the properly you need)
> 
> ![2022-05-13_11-48-09](https://user-images.githubusercontent.com/64448768/168247290-741625f4-3200-4c1e-8147-4edff134767e.jpg) ![2022-05-13_11-48-35](https://user-images.githubusercontent.com/64448768/168247299-0afc774b-db0f-4d41-996e-3c6359c6ea88.jpg) ![2022-05-13_11-48-50](https://user-images.githubusercontent.com/64448768/168247301-75d21354-b38e-473c-b2e8-d8f5107ab2c2.jpg)
> 
> ############################# Dig result as following 👍
> 
> ```python
>  dig sigfail.verteiltesysteme.net @127.0.0.1 -p 5335 (Unbound dns)
> 
> ; <<>> DiG 9.18.1-1ubuntu1-Ubuntu <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 5335
> ;; global options: +cmd
> ;; Got answer:
> ;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 12068
> ;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1
> 
> ;; OPT PSEUDOSECTION:
> ; EDNS: version: 0, flags:; udp: 1232
> ;; QUESTION SECTION:
> ;sigfail.verteiltesysteme.net.  IN      A
> 
> ;; Query time: 412 msec
> ;; SERVER: 127.0.0.1#5335(127.0.0.1) (UDP)
> ;; WHEN: Fri May 13 11:42:42 EEST 2022
> ;; MSG SIZE  rcvd: 57
> 
>  
> 
>  dig sigfail.verteiltesysteme.net @127.0.0.1 -p 5053
> 
> ; <<>> DiG 9.18.1-1ubuntu1-Ubuntu <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 5053
> ;; global options: +cmd
> ;; Got answer:
> ;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 56216
> ;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1
> 
> ;; OPT PSEUDOSECTION:
> ; EDNS: version: 0, flags:; udp: 1232
> ; COOKIE: e48ef842b2bf56b5 (echoed)
> ;; QUESTION SECTION:
> ;sigfail.verteiltesysteme.net.  IN      A
> 
> ;; Query time: 12 msec
> ;; SERVER: 127.0.0.1#5053(127.0.0.1) (UDP)
> ;; WHEN: Fri May 13 11:42:51 EEST 2022
> ;; MSG SIZE  rcvd: 69
> 
> 
> 
>  dig sigfail.verteiltesysteme.net @127.0.0.1 -p 6053  (dnscrypt-proxy)
> 
> 
> ; <<>> DiG 9.18.1-1ubuntu1-Ubuntu <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 6053
> ;; global options: +cmd
> ;; Got answer:
> ;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 61243
> ;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1
> 
> ;; OPT PSEUDOSECTION:
> ; EDNS: version: 0, flags:; udp: 1232
> ;; QUESTION SECTION:
> ;sigfail.verteiltesysteme.net.  IN      A
> 
> ;; Query time: 44 msec
> ;; SERVER: 127.0.0.1#6053(127.0.0.1) (UDP)
> ;; WHEN: Fri May 13 11:42:54 EEST 2022
> ;; MSG SIZE  rcvd: 57
> ```

@jo20201 what os and hardware are you using ? .. you get  alot of errors for Unbound . It should look like this
![image](https://user-images.githubusercontent.com/18756975/168288954-b2211f8a-3eb2-4839-9358-f5f77e49c5f1.png)


.. I know the fix for the warning "warning: so-rcvbuf 1048576" . You need to open `sudo nano /etc/sysctl.conf` and add
```
net.core.rmem_max=1048576
```
 I get that error on a VM Ubuntu, not sure about Pi at the moment(waiting on power supply).. those other errors are weird

**Comment by trinib** [comment]
> > > for me i like to share my result unbound , cloudflare-proxy , dnscrypt-proxy working fine (form me i use family upstream you can use the properly you need)
> > > ![2022-05-13_11-48-09](https://user-images.githubusercontent.com/64448768/168247290-741625f4-3200-4c1e-8147-4edff134767e.jpg) ![2022-05-13_11-48-35](https://user-images.githubusercontent.com/64448768/168247299-0afc774b-db0f-4d41-996e-3c6359c6ea88.jpg) ![2022-05-13_11-48-50](https://user-images.githubusercontent.com/64448768/168247301-75d21354-b38e-473c-b2e8-d8f5107ab2c2.jpg)
> > > ############################# Dig result as following 👍
> > > ```python
> > >  dig sigfail.verteiltesysteme.net @127.0.0.1 -p 5335 (Unbound dns)
> > > 
> > > ; <<>> DiG 9.18.1-1ubuntu1-Ubuntu <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 5335
> > > ;; global options: +cmd
> > > ;; Got answer:
> > > ;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 12068
> > > ;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1
> > > 
> > > ;; OPT PSEUDOSECTION:
> > > ; EDNS: version: 0, flags:; udp: 1232
> > > ;; QUESTION SECTION:
> > > ;sigfail.verteiltesysteme.net.  IN      A
> > > 
> > > ;; Query time: 412 msec
> > > ;; SERVER: 127.0.0.1#5335(127.0.0.1) (UDP)
> > > ;; WHEN: Fri May 13 11:42:42 EEST 2022
> > > ;; MSG SIZE  rcvd: 57
> > > 
> > >  
> > > 
> > >  dig sigfail.verteiltesysteme.net @127.0.0.1 -p 5053
> > > 
> > > ; <<>> DiG 9.18.1-1ubuntu1-Ubuntu <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 5053
> > > ;; global options: +cmd
> > > ;; Got answer:
> > > ;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 56216
> > > ;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1
> > > 
> > > ;; OPT PSEUDOSECTION:
> > > ; EDNS: version: 0, flags:; udp: 1232
> > > ; COOKIE: e48ef842b2bf56b5 (echoed)
> > > ;; QUESTION SECTION:
> > > ;sigfail.verteiltesysteme.net.  IN      A
> > > 
> > > ;; Query time: 12 msec
> > > ;; SERVER: 127.0.0.1#5053(127.0.0.1) (UDP)
> > > ;; WHEN: Fri May 13 11:42:51 EEST 2022
> > > ;; MSG SIZE  rcvd: 69
> > > 
> > > 
> > > 
> > >  dig sigfail.verteiltesysteme.net @127.0.0.1 -p 6053  (dnscrypt-proxy)
> > > 
> > > 
> > > ; <<>> DiG 9.18.1-1ubuntu1-Ubuntu <<>> sigfail.verteiltesysteme.net @127.0.0.1 -p 6053
> > > ;; global options: +cmd
> > > ;; Got answer:
> > > ;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 61243
> > > ;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1
> > > 
> > > ;; OPT PSEUDOSECTION:
> > > ; EDNS: version: 0, flags:; udp: 1232
> > > ;; QUESTION SECTION:
> > > ;sigfail.verteiltesysteme.net.  IN      A
> > > 
> > > ;; Query time: 44 msec
> > > ;; SERVER: 127.0.0.1#6053(127.0.0.1) (UDP)
> > > ;; WHEN: Fri May 13 11:42:54 EEST 2022
> > > ;; MSG SIZE  rcvd: 57
> > > ```
> > 
> > 
> > @jo20201 what os and hardware are you using ? .. you get  alot of errors for Unbound . It should look like this
> > ![image](https://user-images.githubusercontent.com/18756975/168288954-b2211f8a-3eb2-4839-9358-f5f77e49c5f1.png)
> > .. I know the fix for the warning "warning: so-rcvbuf 1048576" . You need to open `sudo nano /etc/sysctl.conf` and add
> > ```
> > net.core.rmem_max=1048576
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I get that error on a VM Ubuntu, not sure about Pi at the moment(waiting on power supply).. those other errors are weird
> 
> Thx dude now the issue fixed with your help and I disable the log ![Screenshot_20220513-162908_JuiceSSH.jpg](https://user-images.githubusercontent.com/64448768/168294266-63d4d9b5-b679-4b47-87cd-06fdd5e56ba2.jpg)

What are you running Unbound on ? I still see these messages "notice: init module..."

**Comment by trinib** [comment]
> this error in config file for unbound related with
> 
> ```python
>  # Ensure kernel buffer is large enough to not lose messages in traffix spikes
>     so-rcvbuf: 4m
>     so-sndbuf: 4m 
> ```

@jo20201 yea your right. 4m seems to work fine on PI. I do not know exactly why that happens in Ubuntu. you still have not said what are you running it on.

There is another way I figured out. You can set it in `unbound.conf` in kb. For example if it shows :

![image](https://user-images.githubusercontent.com/18756975/168337352-28da758d-7f46-4c08-acb9-e1f27afa0879.png)

Open `sudo nano /etc/unbound/unbound.conf.d/unbound.conf` and set 
![image](https://user-images.githubusercontent.com/18756975/168337389-90c766ac-1507-45a0-a861-8d57598abc24.png)

Restart service:
```
sudo systemctl restart unbound
```

and no error

**Comment by trinib** [comment]
> > > this error in config file for unbound related with
> > > ```python
> > >  # Ensure kernel buffer is large enough to not lose messages in traffix spikes
> > >     so-rcvbuf: 4m
> > >     so-sndbuf: 4m 
> > > ```
> > 
> > 
> > @jo20201 yea your right. 4m seems to work fine on PI. I do not know exactly why that happens in Ubuntu. you still have not said what are you running it on.
> > There is another way I figured out. You can set it in `unbound.conf` in kb. For example if it shows :
> > ![image](https://user-images.githubusercontent.com/18756975/168337352-28da758d-7f46-4c08-acb9-e1f27afa0879.png)
> > Open `sudo nano /etc/unbound/unbound.conf.d/unbound.conf` and set ![image](https://user-images.githubusercontent.com/18756975/168337389-90c766ac-1507-45a0-a861-8d57598abc24.png)
> > Restart service:
> > ```
> > sudo systemctl restart unbound
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and no error
> 
> my OS is ubuntu 22.04
> 
> ![2022-05-13_20-48-59](https://user-images.githubusercontent.com/64448768/168339661-c53e258d-85c4-45c5-80f1-837ccb170b86.jpg)

@jo20201 what hardware ? no VM right ? if using PI, I guess its a Ubuntu thing ..

**Comment by trinib** [comment]
@eltonajmenezes i have been using it for a while and issue has not arrive for me.
